app/src/main/java/io/github/jbellis/brokk/ContextManager.java

```diff
@@ -747,27 +746,12 @@ public void dropAll() {
 
     /** Drop fragments by their IDs. */
     public void drop(Collection<? extends ContextFragment> fragments) {
-        var ids = fragments.stream().map(f -> mapToLiveFragment(f).id()).toList();
+        var ids = fragments.stream().map(f -> contextHistory.mapToLiveFragment(f).id()).toList();
         pushContext(currentLiveCtx -> currentLiveCtx.removeFragmentsByIds(ids));
         io.systemOutput("Dropped "
                 + fragments.stream().map(ContextFragment::shortDescription).collect(Collectors.joining(", ")));
     }
 
-    /**
-     * Occasionally you will need to determine which live fragment a frozen fragment came from. This does that by
-     * assuming that the live and frozen Contexts have their fragments in the same order.
-     */
-    private ContextFragment mapToLiveFragment(ContextFragment f) {
-        if (!(f instanceof FrozenFragment)) {
-            return f;
-        }
-
-        var ctx = topContext();
-        int idx = ctx.getAllFragmentsInDisplayOrder().indexOf(f);
-        assert idx >= 0 : "Fragment %s not found in top context %s".formatted(f, ctx.getAllFragmentsInDisplayOrder());
-        return liveContext().getAllFragmentsInDisplayOrder().get(idx);
```

Comment:
It was possible that changes to the top and live context happen after calling `topContext()` but before `liveContext()` so index would point to the wrong fragment. Thus moved to a synchronized method in `ContextHistory`.